### PR TITLE
I76 parser

### PR DESCRIPTION
**Pull request by weiym97**
This is a pull request from i76-parser to the main branch.

Basically this new branch solves the four schemas provided by Chris.

#### Schema 1
The configuration file is [RejectCase.json](./schema/RejectCase.json)
```
if patient_name matches ' d{6}': // e.g. patient name is 000123
insert patient_name into clinical_trial_subject_id
else:
reject image
```

#### Schema 2
The configuration file is [MultipleActionCase.json](./schema/MultipleActionCase.json)
```
if clinical_trial_protocol_id matches '[A Z]{3} d{6}': // e.g. ABC123456
prepend patient_name with "_"
prepend patient_name with clinical_trial_protocol_id // patient_name is now ABC123456_000123
else:
reject image
```

#### Schema 3
The configuration file is [DateoverwriteCase.json](./schema/DateoverwriteCase.json)
```
if date_of_birth matches ' d{8}': // e.g. 19550531
date_of_birth replace dd with 30
date_of_birth replace mm with 06
```

#### Schema 4
The configuration file is [MultipleConditionCase.json](./schema/MultipleConditionCase.json)
```
if modality == 'MR':
if manufacturer is 'SIEMENS' or 'HITACHI':
if magnetic_field_strength == 1.5
and repetition_time between 2000 and 2200
and echo_time lessthan 50
and inversion_time between 1000 and 1200
and mr_acquisition_type is '3D':
set clinical_trial_series_description to "T1"
else if magnetic_field_strength == 3.0
and repetition_time between 2000 and 2200
and echo_time lessthan 50
and inversion_time between 1000 and 1200
and mr_acquisition_type is '3D':
set clinical_trial_series_description to "T1"
```

Grossly speaking, all these schemas are implemented now, but I'm not 100% satisfied. Here are some points:
(1) **Schema 2**. Do not have control over the order of actions (COPY, PREPEND), seems that COPY always comes first now (parsed in alphabetical order).
(2) **Schema 3**. Only check it's an 8-digit string, never know whether the date is valid (what about `19990229`).
(3) **Schema 4**. ```AND``` and ```OR``` usually do not work well because we have replicated keys (we usually have A equals 100 and B equals 10, etc), and only one of them will be parsed. Also, those comes after ```IF_TRUE``` or ```IF_FALSE``` alphabetically will be ignored (because the parsing is done in alphabetical order). One compromise is to break up AND/OR into nested conditions (what we did in schema 4). It usually works fine but requires long json files (~70 lines for our simplified version of schema 4, might >150 lines for the whole schema).

### CODE REVIEW

File: src/python_plugin/prototype.py
Comment (ian-mcfarlane) [inline src/python_plugin/prototype.py:1]: What is this file for?

**Review by ian-mcfarlane — CHANGES_REQUESTED** [review]
- python_plugin/prototype.py is empty - what is this for?
- unit tests and sanitiser checks have failed - any idea why?
Great work on these test cases and updating the parser!

File: src/python_plugin/prototype.py
Comment (weiym97) [reply src/python_plugin/prototype.py:1]: Sorry, this file was originally for i75. I just forgot to delete it.

**Review by martinjrobins — CHANGES_REQUESTED** [review]
looks great, just remove the commented out code, thanks @weiym97

**Pull request merged**